src/common/pip.ts

```diff
@@ -68,22 +70,39 @@ export const runPipList = async (onStdio?: OnStdio): Promise<PipList> => {
 export const runPipInstall = async (
     dependencies: readonly Dependency[],
     onProgress?: (percentage: number) => void,
-    onStdio?: OnStdio
+    onStdio?: OnStdio,
+    noProgress = false
 ): Promise<void> => {
     onProgress?.(0);
-    // TODO: implement progress
-    await runPip(
-        ['install', '--upgrade', ...dependencies.map((d) => `${d.packageName}==${d.version}`)],
-        onStdio
-    );
+    if (noProgress) {
+        // TODO: implement progress via this method (if possible)
+        const deps = dependencies
+            .map((d) => d.packages.map((p) => `${p.packageName}==${p.version}`))
+            .flat();
+        await runPip(['install', '--upgrade', ...deps], onStdio);
+    } else {
+        for (const dep of dependencies) {
+            for (const pkg of dep.packages) {
+                // eslint-disable-next-line no-await-in-loop
+                const { python } = await getPythonInfo();
```

Comment:
Well I should definitely move the python check out of there, yeah. But the one below needs to be there. Eslint wants you to use promise.all to run multiple promises, but I need them to run synchronously rather than asynchronously, so therefore await in the loop.